neo/SmartContract/ApplicationEngine.cs

```diff
@@ -19,6 +19,7 @@ public partial class ApplicationEngine : ExecutionEngine
         private readonly List<NotifyEventArgs> notifications = new List<NotifyEventArgs>();
         private readonly List<IDisposable> disposables = new List<IDisposable>();
 
+        public Dictionary<UInt160, int> InvocationCounter { get; } = new Dictionary<UInt160, int>();
```

Comment:
```suggestion
        public IReadOnlyDictionary<UInt160, int> InvocationCounter { get; } = new Dictionary<UInt160, int>();
```

Reply:
We need to be able to change the dictionary